src/WooCommerce.php

```diff
@@ -380,14 +385,19 @@ public static function getProductDetailsHtmlDataAttr(\WC_Product $product, $is_d
     $product_id = $product->get_id();
     $product_details = static::getProductDetails($product_id);
 
-    if ('variable' === $product->get_type() && $is_detail_view) {
+    if (($product->get_type() === 'variable' && $is_detail_view) || ($product->get_type() === 'variation' && !isset($product_details['variant']))) {
       $attributes = $product->get_variation_attributes();
       $selected_attributes = [];
       foreach ($attributes as $attribute_name => $options) {
-        $attribute = isset($_REQUEST['attribute_' . sanitize_title($attribute_name)]) ? wc_clean(stripslashes(urldecode($_REQUEST['attribute_' . sanitize_title($attribute_name)]))) : '';
-        if ($attribute) {
-          $attribute_term_data = get_term_by('slug', $attribute, $attribute_name);
-          $selected_attributes[] = $attribute_term_data ? $attribute_term_data->name : $attribute;
+        if ($product->get_type() === 'variable') {
+          $attribute = isset($_REQUEST['attribute_' . sanitize_title($attribute_name)]) ? wc_clean(stripslashes(urldecode($_REQUEST['attribute_' . sanitize_title($attribute_name)]))) : '';
+          if ($attribute) {
+            $attribute_term_data = get_term_by('slug', $attribute, $attribute_name);
+            $selected_attributes[] = $attribute_term_data ? $attribute_term_data->name : $attribute;
+          }
+        }
+        if ($product->get_type() === 'variation' && !$selected_attributes) {
```

Comment:
Why do we only set the options of the first attribute we find?
```suggestion
        if ($product->get_type() === 'variation') {
```

If you intended this as it is, then why are we retrieving and looping over all attributes instead of just a single one in the first place? 🤔